core/src/main/scala/org/apache/spark/internal/config/Deploy.scala

```diff
@@ -82,4 +84,12 @@ private[spark] object Deploy {
     .checkValue(_ > 0, "The maximum number of running drivers should be positive.")
     .createWithDefault(Int.MaxValue)
 
+  val DRIVER_ID_PATTERN = ConfigBuilder("spark.deploy.driverIdPattern")
+    .doc("The pattern for driver ID generation based on Java `String.format` method. " +
+      "The default value is `driver-%s-%04d` which represents the existing driver id string " +
+      ", e.g., `driver-20231031224459-0019`. Please be careful to generate unique IDs")
+    .version("4.0.0")
+    .stringConf
+    .checkValue(!_.format(new Date(), 0).exists(_.isWhitespace), "Whitespace is not allowed.")
```

Comment:
We execute `String.format` here in advance to prevent the failure in `Master` class.